File: src/contests/cg/puzzle/perl/distributing-candy.pl
# 🎮 CodinGame Puzzle - distributing-candy
# https://www.codingame.com/training/easy/distributing-candy

my ($n, $m) = split ' ', <STDIN>;
my @candies = sort { $a <=> $b } split ' ', <STDIN>;
my $best = 9**9**9;
for my $i (0 .. $n - $m) {
    my $diff = $candies[$i + $m - 1] - $candies[$i];
    $best = $diff if $diff < $best;
}
print "$best\n";

File: src/contests/cg/puzzle/dart/distributing-candy.dart
// 🎮 CodinGame Puzzle - distributing-candy
// https://www.codingame.com/training/easy/distributing-candy

import 'dart:io';

void main() {
  var first = stdin.readLineSync()!.trim().split(' ');
  int n = int.parse(first[0]);
  int m = int.parse(first[1]);
  List<int> candies =
      stdin.readLineSync()!.trim().split(' ').map(int.parse).toList();
  candies.sort();
  int best = 1 << 62;
  for (int i = 0; i + m - 1 < n; i++) {
    int diff = candies[i + m - 1] - candies[i];
    if (diff < best) best = diff;
  }
  print(best);
}
